articles/nodejs-ejs/index.md

```diff
@@ -0,0 +1,207 @@
+- ```ejs
+   <% posts.forEach((post) => { %>
+                    <a href="#" class="list-group-item">
+                        <h4 class="list-group-item-heading"><%= post.title %></h4>
+                        <p class="list-group-item-text"><%= post.author %></p>
+                    </a>
+                    <br>
+                    <br>
+                <% }) %>
+    ```
+    The above code snippet loops through the `posts` array that will be passed from the `IndexRoute` file. Foreach `post`, a title, and the author name displays within the posts div.
+
+
+### Routes
+In the routes folder, we store the javascript files containing the logic that is executed whenever a certain route is visited. In our case whenever you visit `121.0.0.1:5000/` when the development server is running, the logic in our `IndexRoute.js` file will be executed to respond to the request.
+
+In the root project directory, create a file named `IndexRouter.js` and add the code snippets below into it.
+   
+
+```javascript
+const express = require('express');
+const router = express.Router();
+const posts = [
+
+    {
+        id: 1,
+        author: 'John',
+        title: 'Templating with pug',
+        body: 'Blog post 1'
+    },
+
+    {
+        id: 2,
+        author: 'Peter',
+        title: 'React: Starting from the Bottom',
+        body: 'Blog post 2'
+    },
+
+    {
+        id: 3,
+        author: 'Violet',
+        title: 'Node.js Streams',
+        body: 'Blog post 3'
+    },
+
+    {
+        id: 4,
+        author: 'Condy',
+        title: 'Node.js Events',
+        body: 'Blog post 4'
+    }
+
+]
+
+/* GET home page. */
+// Returns a list of posts to the view.
+router.get('/', function (req, res, next) {
+    res.render('index', {title: 'Express', posts: posts});
+});
+
+module.exports = router;
+```
+In the above code snippet, we create an array of `posts` that we will display whenever a user visits the route `/`. 
+
+`module.exports = router;` exports the file so that we can import it in the `app.js` file.
+
+### App.js
+All configurations and the main application routes go into this file.
+```javascript
+const express = require('express');
+const indexRouter = require('./routes/index');
+const app = express();
+
+// view engine setup
+app.set('views', path.join(__dirname, 'views'));
+app.set('view engine', 'ejs');
+app.use('/', indexRouter);
+
+app.listen(3000)
+console.log('listening on port 3000')
+```
+- `const app = express();` creates an instance of the express, making our application an express application.
+- `app.set('views', path.join(__dirname, 'views'));` sets the view directory to `views`. This is where all our `EJS` files will be stored.
+- `app.set('view engine', 'ejs');` sets the view engine to `EJS`.
+- `app.listen(3000)` starts the development server.
+
+start the server by executing the command below.
+```bash
+$ node app.js
+```
+When we navigate to `localhost:3000/` in our browser, a page with all the posts is displayed.
```

Comment:
Consider changing "When we navigate to `localhost:3000/` in our browser, a page with all the posts is displayed." to "When we navigate to `localhost:3000/` in our browser, a page with all the posts is displayed as shown below."